### ISSUE TITLE
DaemonSet pods are scheduled onto control plane nodes by default

### ISSUE THREAD

**Issue by itspngu** [kubernetes-sigs/secrets-store-csi-driver#1177]
**What steps did you take and what happened:**
When installing the CSI driver using the official Helm chart, the `Daemonset.spec.template.spec.tolerations` are set to `[{ operator: Exists }]`, which leads to driver pods being scheduled to all nodes, including those hosting the kubernetes control plane.

**What did you expect to happen:**
No driver pods are scheduled to control plane nodes.

**Anything else you would like to add:**
The tolerations can be overridden using `values.yaml`, in fact even a [comment explaining the setting](https://github.com/kubernetes-sigs/secrets-store-csi-driver/blob/main/charts/secrets-store-csi-driver/values.yaml#L84-L87) is included:

```yaml
# ref: https://kubernetes.io/docs/concepts/scheduling-eviction/taint-and-toleration/
# An empty key with operator Exists matches all keys, values and effects which means this will tolerate everything.
tolerations:
  - operator: "Exists"
```

Is there a reason for this being the default? A quick test I've performed seems to indicate that running the driver on control plane nodes is not required (unless you have other workloads on these nodes which you've configured to consume CSI secrets, which shouldn't be the norm as far as I can imagine).

**Which provider are you using:**
n/a

**Environment:**

- Secrets Store CSI Driver version: (use the image tag): n/a
- Kubernetes version: (use `kubectl version`): n/a

**Comment by aramase** [comment]
> Is there a reason for this being the default? A quick test I've performed seems to indicate that running the driver on control plane nodes is not required (unless you have other workloads on these nodes which you've configured to consume CSI secrets, which shouldn't be the norm as far as I can imagine).

The change was done recently to ensure the driver pods run on every node by default. Prior to this change, lots of users were running into issues where new nodes were added with taints that the driver didn't tolerate by default and scaling workloads would fail without the driver running. 

The `tolerations` value can be overridden, so you can change the defaults when deploying in your cluster.

**Comment by aayushharwani** [comment]
Isn’t it better to follow the standard and keep the default as no tolerations, since that’s what is followed across other DaemonSets as well?

While adding Fargate profiles to my EKS cluster, I noticed that only the secrets-store-csi-driver DaemonSet pods were being created on Fargate nodes. Later, I realized this was due to the default toleration value.

I think it’s better to keep it empty by default, so users consciously override it when needed.
